performance_test/src/data_running/data_runner_base.hpp

```diff
@@ -39,7 +44,28 @@ class DataRunnerBase : boost::noncopyable
 public:
   DataRunnerBase()
   : m_ec(ExperimentConfiguration::get())
-  {}
+  {
+    if (m_ec.check_memory()) {
+#ifdef MEMORY_TOOLS_ENABLED
+      osrf_testing_tools_cpp::memory_tools::initialize();
+
+      auto on_unexpected_memory =
```

Comment:
`auto const on_unexpected_memory`

Reply:
For my own education:  

What happens with this lambda when you get out of scope?

Should not you use a plain function callback in such a  case?

Reply:
The lambda is wrapped in a https://en.cppreference.com/w/cpp/utility/functional/function. So this should fine.

Reply:
The function is now also const, as requested and correct.